### ISSUE TITLE
Add Simple Property System to Demo App

### ISSUE THREAD

**Issue by julianeisel** [julianeisel/bWidgets#7]
The way data is manipulated after interactions in the demo application is quite ugly and not representable for how a normal application would do this. It's simply a bad/misleading demo.
For every widget that's supposed to modify data in some way, we have to add a new functor class (!), create an instance of it and assign it to `bwWidget.apply_functor()`. However, most of these classes do very similar things which could be generalized with the help of a simple property system.

The following features are needed:
* Defining properties of different types (booleans, floats, enumerations, etc.).
* Setting min/max values of numeric type properties.
* Custom set and/or update callbacks for properties.
* Maybe useful to test tooltips in the future: Setting a description for each property.
So basically a quite simplified version of RNA.

Then we can add generic functor classes to manipulate these. Also, we can try how screen-graph builders specifically for such properties would work. This idea seems to have quite some potential.
